spec/acceptance/basic_dsc_resources/service/service_spec.rb

```diff
@@ -0,0 +1,128 @@
+require 'spec_helper_acceptance'
+
+describe 'Service tests' do
+
+  context 'MODULES-2336 - C68749 - Attempt to Apply DSC Service Resource with Incompatible "StartupType" and "State"' do
+
+    # ERB Manifest
+    dsc_type = 'service'
+    dsc_props = {
+      :dsc_name        => 'w32time',
+      :dsc_state       => 'Running',
+      :dsc_startuptype => 'Disabled'
+    }
+
+    dsc_manifest = single_dsc_resource_manifest(dsc_type, dsc_props)
+
+    # Verify
+    error_msg = /Error\:.*Dsc_service\[service_test\]:.*Cannot start and disable a service/m
+
+    windows_agents.each do | agent |
+      it 'Should apply the manifest and encounter an error' do
+        on(agent, puppet('apply'), :stdin => dsc_manifest, :acceptable_exit_codes => [0,2]) do |result|
+          assert_match(error_msg, result.stderr, 'Expected error was not detected!')
+        end
+      end
+    end
+  end
+
+  context 'MODULES-2336 - C68748 - Attempt to Apply DSC Service Resource with Invalid "Name" Specified' do
+
+    # ERB Manifest
+    dsc_type = 'service'
+    dsc_props = {
+      :dsc_name        => 'makebelieve',
+      :dsc_state       => 'Running',
+      :dsc_startuptype => 'Automatic'
+    }
+
+    dsc_manifest = single_dsc_resource_manifest(dsc_type, dsc_props)
+
+    error_msg = /Error: PowerShell DSC resource MSFT_ServiceResource.*Dsc_service\[service_test\].*The service 'makebelieve' does not exist/m
+
+    windows_agents.each do | agent |
+      it 'Should apply the manifest and encounter an error.' do
+        on(agent, puppet('apply'), :stdin => dsc_manifest, :acceptable_exit_codes => [0,2]) do |result|
+          assert_match(error_msg, result.stderr, 'Expected error was not detected!')
+        end
+      end
+    end
+  end
+
+  context 'MODULES-2511 - C68729 - Apply DSC Service Resource that Removes a Service' do
+    # ERB Manifest
+    dsc_type = 'service'
+    dsc_module = 'PSDesiredStateConfiguration'
+    dsc_props = {
+      :dsc_ensure => 'Absent',
+      :dsc_name   => 'puppet'
+    }
+
+    dsc_manifest = single_dsc_resource_manifest(dsc_type, dsc_props)
+
+    windows_agents.each do | agent |
+      it 'Should apply a manifest with no errors.' do
+        on(agent, puppet('apply'), :stdin => dsc_manifest, :acceptable_exit_codes => [0,2]) do |result|
+          assert_no_match(/Error:/, result.stderr, 'Unexpected error was detected!')
+        end
+      end
+
+      it 'Should verify results' do
+        expect_failure('Expected to fail because of MODULES-2512') do
```

Comment:
MODULES-2512 is resolved as fixed, so this either should work, or should fail for a different reason from MODULES-2512

Reply:
Checking.

Reply:
The fix for MODULES-2512 was a [documentation fix only](https://github.com/puppetlabs/puppetlabs-dsc/commit/5e8873957f1ff528ee0975bd1a401f0f822e39e8#diff-04c6e90faac2675aa89e2176d2eec7d8R348). The test still passes because the manifest constructed and passed into `assert_dsc_resource` does not follow the guidelines recommended in the docs change. In effect, this is a canary test that will simply tell us if MS ever fixes this bug.

Reply:
Word, makes sense.